code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm

```diff
@@ -111,33 +111,33 @@
 	else if(W.isscrewdriver() && canbemoved)
 		if(screwed)
 			to_chat(user,  "<span class='notice'>You start to unscrew the locker from the floor...</span>")
-			playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
-			if (do_after(user, 10 SECONDS, act_target = src))
+			playsound(loc, W.usesound, 50, 1)
+			if (do_after(user, 10/W.toolspeed, act_target = src))

```

Comment:
This change will make it happen after 10 ticks / toolspeed instead of 10 seconds / toolspeed